File: app/src/sharedTest/java/org/watsi/uhp/BillableFactory.java
package org.watsi.uhp;

import org.watsi.uhp.models.Billable;

import java.sql.SQLException;

public class BillableFactory {
    public static Billable createBillable(String name, Billable.TypeEnum type, String unit,
              String composition, int price, boolean requiresLabResult) throws SQLException {
        Billable billable = new Billable();
        billable.generateId();
        billable.setName(name);
        billable.setType(type);
        billable.setUnit(unit);
        billable.setComposition(composition);
        billable.setPrice(price);
        billable.setCreatedDuringEncounter(false);
        billable.setRequiresLabResult(requiresLabResult);
        billable.create();
        return billable;
    }

    public static Billable createBillable(String name, Billable.TypeEnum type, int price,
              boolean requiresLabResult) throws SQLException {
        Billable billable = new Billable();
        billable.generateId();
        billable.setName(name);
        billable.setType(type);
        billable.setPrice(price);
        billable.setCreatedDuringEncounter(false);
        billable.setRequiresLabResult(requiresLabResult);
        billable.create();
        return billable;
    }
}

File: domain/src/test/java/org/watsi/domain/factories/BillableFactory.kt
package org.watsi.domain.factories

import org.watsi.domain.entities.Billable
import java.util.UUID

object BillableFactory {

    fun build(
            id: UUID = UUID.randomUUID(),
            type: Billable.Type = Billable.Type.SERVICE,
            composition: String? = null,
            unit: String? = null,
            name: String = "Delivery") : Billable {
        return Billable(id = id,
                        type = type,
                        composition = composition,
                        unit = unit,
                        name = name)
    }
}
